packages/cli/src/command-build.js

```diff
@@ -502,30 +495,71 @@ function makePackageBuild(
  * @param {import('@bilt/types').Directory} packageDirectory
  * @param {Record<string, string | boolean | undefined>} buildOptions
  * @param {object} biltin
- * @param {(se: import('@bilt/build-with-configuration').StepExecution) => void} logExecution
  */
-async function executePhase(
-  jobConfiguration,
-  phase,
-  packageDirectory,
-  buildOptions,
-  biltin,
-  logExecution,
-) {
+async function executePhase(jobConfiguration, phase, packageDirectory, buildOptions, biltin) {
   const stepExecutions = getPhaseExecution(
     jobConfiguration.steps[phase],
     packageDirectory,
     buildOptions,
     {directory: packageDirectory, biltin},
   )
+  return new Listr(
+    [
+      {
+        title: phase,
+        task: (_, task) => task.newListr(convertStepExecutionsToTasks(stepExecutions)),
+        options: {
+          persistentOutput: true,
+        },
+      },
+    ],
+    {
+      rendererOptions: {
+        showTimer: true,
+        collapseErrors: false,
+      },
+    },
+  ).run()
+}
+
+/** @type {( linesToDisplay?: number) => (payload: string, showAll?: boolean) => string} */
+const StreamBuffer = (linesToDisplay = 10) => {
+  /** @type {string[]} */
+  let buffer = []
+  return (payload, showAll = false) => {
+    const newLines = splitLines(payload.toString())
+    buffer = buffer.concat(newLines)
+    if (showAll) return buffer.join('\n')
+    return buffer.slice(Math.max(0, buffer.length - linesToDisplay), buffer.length).join('\n')
+  }
+}
 
-  for (const stepExecution of stepExecutions.filter((se) => se.isEnabled())) {
-    if (await stepExecution.shouldSkip()) {
+/**
+ * @param {import('@bilt/build-with-configuration').StepExecution[]} stepExecutions
+ * @returns {import('listr2').ListrTask[]}
+ */
+function convertStepExecutionsToTasks(stepExecutions) {
+  return stepExecutions.map((stepExecution) => ({
+    title: stepExecution.info().name,
+    task: async (_, task) => {
       const childProcess = await stepExecution.executeToChildProcess()
-      childProcess.stdout.pipe(process.stdout)
-      childProcess.stderr.pipe(process.stderr)
+      const buffer = StreamBuffer()
+      childProcess.stdout.on('data', (payload) => {
```

Comment:
It won't clear it, it will append to it they share the same buffer.
I think it'll be easier to handle issues if you see the entire log of the process.

Reply:
Right! One array for both. Missed that. Thanks!